File: src/main/resources/frontend/tcpproject/src/app/models/channel.ts
import { Subscription } from "./subscription";

export class Channel {
  id: number;
  name: string;
  direct: boolean;
  subscriptions: Subscription[];

  selectedChannel: number;

  constructor(){
    this.id = 2;
    this.name = "default;"
  }

  updateId(idNumber :number) {
    this.id=idNumber;
  }
}

File: src/main/java/com/TCPGroup/project/Models/Channel.java
package com.TCPGroup.project.Models;

import javax.persistence.*;
import java.util.List;

@Entity
public class Channel {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    private String name;
    private boolean direct;

    //@OneToMany(fetch = FetchType.LAZY, mappedBy = "channel")
    @OneToMany(fetch = FetchType.LAZY, mappedBy = "channelId")
    private List<Subscription> subscriptions;

    public Channel(){ }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isDirect() {
        return direct;
    }

    public void setDirect(boolean direct) {
        this.direct = direct;
    }

    public List<Subscription> getSubscriptions() { return subscriptions; }

    public void setSubscriptions(List<Subscription> subscriptions) { this.subscriptions = subscriptions; }
}
